### ISSUE TITLE
Empty User Defined Table Type Args

### ISSUE THREAD

**Issue by ProgrammingEvangelist** [jonwagner/Insight.Database#215]
We are currently running *Insight.Database* version number:  `5.0.1`.

We are starting to see an increasing number of stored procedure calls that are invoked with an empty user-defined table type.  It took us a very long time to determine that this was the case.  Both stored proc's that we have witnessed this occurring on expect a UDT.  We have logging in place to ensure that just before we pass the items into *Insight.Database* call(s), they are populated lists containing data.  However, the **SQL Server Profiler** clearly shows that the UDT, while declared -- is never inserted into with the aforementioned data we know we have passed, thus it is empty.  

For example:

```sql
declare @p3 dbo.udt_PointDateRange

exec spu_GetHighLevelForecastPointValues @customerId='4A8CE18A-73AA-43DC-86CE-66FFBECA5B2D',@plantObjectId='FE4B1991-BD1E-4A53-B34A-8E8141D693CE',@pointsWithDateRange=@p3,@futureOnly=0,@pastOnly=1
```

The two *Insight.Database* calls where this has been confirmed to occur are:

```c#
InsertList(string sql, IEnumerable<TResult> inserted)
```
and
```c#
QueryAsync<T>(string sql, object parameters, int commandTimeout)
```

Our DBA discovered that she was able to reproduce the issue twice.  While profiling for performance, as soon as the `RPC Completed` SQL event triggered and appeared in the profiler, she would immediately click the web UI to perform another request with varied selections (ultimately altering the arguments), this is when *Insight.Database* would pass in an empty list to the stored proc.

**Comment by jonwagner** [comment]
I'd have to debug this one. I can't say when I'll have time to look at it.

**Comment by jonwagner** [comment]
This is fixed in the next build, 5.2.6. I will do a build before I get distracted. :)
